modules/tinymce/src/plugins/template/test/ts/browser/TemplateSanityTest.ts

```diff
@@ -76,64 +77,80 @@ describe('browser.tinymce.plugins.template.TemplateSanityTest', () => {
-    const editor = hook.editor();
-    const unsanitizedHtml = '<img src="error" onerror="throw new Error();">';
-    addSettings({
-      templates: [{ title: 'a', description: 'b', content: unsanitizedHtml }],
-    });
-    const unsanitizedPreviewHtmlSelector = 'p > img[src="error"][onerror="throw new Error();"]';
-    const sanitizedPreviewHtmlSelector = 'p > img[src="error"][data-mce-src="error"]';
-    const assertPreviewContent = (dialogEl: SugarElement<Node>, existsSelector: string, notExistsSelector: string): void => {
-      UiFinder.findIn<HTMLIFrameElement>(dialogEl, 'iframe').fold(
-        () => assert.fail('Preview iframe not found'),
-        (iframe) => {
-          // fallback for pre-IE 8 using contentWindow.document
-          const iframeDoc = iframe.dom.contentDocument || iframe.dom.contentWindow?.document;
-          const iframeBody = SugarElement.fromDom(iframeDoc?.body as Node);
-          UiFinder.exists(iframeBody, existsSelector);
-          UiFinder.notExists(iframeBody, notExistsSelector);
-        }
-      );
+  context('Previewing unparsed content', () => {
+    const unparsedHtml = '<img src="error" onerror="throw new Error();">';
+    const unparsedPreviewHtmlSelector = 'p > img[src="error"][onerror="throw new Error();"]';
+    const parsedPreviewHtmlSelector = 'p > img[src="error"][data-mce-src="error"]';
+
+    const pPreviewAndAssertNoUnparsedContent = async (editor: Editor): Promise<void> => {
+      const assertNoUnparsedContent = (dialogEl: SugarElement<Node>): void => {
+        UiFinder.findIn<HTMLIFrameElement>(dialogEl, 'iframe').fold(
+          () => assert.fail('Preview iframe not found'),
+          (iframe) => {
+            const iframeDoc = iframe.dom.contentDocument;
+            const iframeBody = SugarElement.fromDom(iframeDoc?.body as Node);
+            UiFinder.exists(iframeBody, parsedPreviewHtmlSelector);
+            UiFinder.notExists(iframeBody, unparsedPreviewHtmlSelector);
+          }
+        );
+      };
+
+      try {
+        await pPreviewTemplate(editor, assertNoUnparsedContent);
+      } catch {
+        assert.fail('Unparsed html read');
+      }
     };
 
-    try {
-      await pPreviewTemplate(editor, (dialogEl) => assertPreviewContent(dialogEl, sanitizedPreviewHtmlSelector, unsanitizedPreviewHtmlSelector));
-    } catch {
-      assert.fail('Unsanitized html read');
-    }
+    it('TINY-9244: Parsed html should be shown when previewing template', async () => {
+      const editor = hook.editor();
+      addSettings({
+        templates: [{ title: 'a', description: 'b', content: unparsedHtml }],
+      });
+      await pPreviewAndAssertNoUnparsedContent(editor);
+    });
+
+    it('TINY-9867: Parsed html should be shown when previewing template containing <html> tags', async () => {
+      const editor = hook.editor();
+      addSettings({
+        templates: [{ title: 'a', description: 'b', content: `<html>${unparsedHtml}</html>` }],
```

Comment:
What if unparsedHtml contains html tag(s)?

Reply:
Our parser would handle any duplicated / malformed tags in that case.